tracing-subscriber/src/fmt/fmt_layer.rs

```diff
@@ -551,6 +627,37 @@ where
     }
 }
 
+struct Timings {
+    accumulated: u64,
+    entered: Option<Instant>,
+}
+
+impl Timings {
+    fn new() -> Self {
+        Self {
+            accumulated: 0,
+            entered: Some(Instant::now()),
+        }
+    }
+    fn get_accumulated(&self) -> String {
+        if self.accumulated >= 10_000_000_000 {
+            format!("{}s", self.accumulated)
+        } else if self.accumulated >= 1_000_000_000 {
+            format!("{:.1}s", self.accumulated as f64 / 1_000_000_000.0)
+        } else if self.accumulated >= 10_000_000 {
+            format!("{}ms", self.accumulated / 1_000_000)
+        } else if self.accumulated >= 1_000_000 {
+            format!("{:.1}ms", self.accumulated as f64 / 1_000_000.0)
+        } else if self.accumulated >= 10_000 {
+            format!("{}µs", self.accumulated / 1000)
+        } else if self.accumulated >= 1000 {
+            format!("{:.1}µs", self.accumulated as f64 / 1000.0)
+        } else {
+            format!("{}ns", self.accumulated)
+        }
```

Comment:
Depending on the feature flags that are enabled, we may already depend on `chrono` --- we _might_ want to outsource this formatting to chrono in that case?

Reply:
Yes, will look into it.